### PR TITLE
Implement query interruption

### PR DESCRIPTION
**Pull request by acquamarin**
This PR implements the query interruption logic.

### CODE REVIEW

File: src/include/main/client_context.h [hunk 1]
```diff
@@ -24,8 +25,14 @@ class ClientContext {
      */
     KUZU_API ~ClientContext() = default;
 
+    /**
+     * @brief Returns whether the current query is interrupted or not.
+     */
+    KUZU_API bool isInterrupted() const { return interrupted; }
```
Comment (ray6080) [inline src/include/main/client_context.h:31]: Do we need to expose this as a public interface?

**Review by andyfengHKU — APPROVED** [review]

File: src/include/main/client_context.h (hunk 1 above)
Comment (acquamarin) [reply src/include/main/client_context.h:31]: Physical operator needs to access this field

File: src/include/common/exception.h [hunk 2]
```diff
@@ -91,5 +91,10 @@ class TransactionManagerException : public Exception {
     explicit TransactionManagerException(const std::string& msg) : Exception(msg){};
 };
 
+class InterruptException : public Exception {
+public:
+    explicit InterruptException() : Exception("User interrupted."){};
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/common/exception.h:96]: done

File: src/include/main/connection.h [hunk 3]
```diff
@@ -131,6 +131,11 @@ class Connection {
     // TODO(Change): move this to C-header once we have C-APIs.
     KUZU_API std::unique_ptr<QueryResult> kuzu_query(const char* queryString);
 
+    /**
+     * @brief interrupts the current executing query.
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/main/connection.h:135]: changed comment

File: test/include/main_test_helper/main_test_helper.h [hunk 4]
```diff
@@ -27,6 +27,12 @@ class ApiTest : public BaseGraphTest {
         ASSERT_EQ(tuple->getValue(0)->getValue<int64_t>(), 8);
         ASSERT_FALSE(result->hasNext());
     }
+
+    static void startLongQuery(Connection* conn) {
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply test/include/main_test_helper/main_test_helper.h:31]: done

File: test/main/connection_test.cpp [hunk 5]
```diff
@@ -137,3 +137,10 @@ TEST_F(ApiTest, Profile) {
                     "b.fName='Farooq' } RETURN a.ID, min(a.age)");
     ASSERT_TRUE(result->isSuccess());
 }
+
+TEST_F(ApiTest, Interrupt) {
+    std::thread longQueryThread(startLongQuery, conn.get());
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply test/main/connection_test.cpp]: renamed

**Pull request merged**